pfcpiface/main.go

```diff
@@ -90,6 +92,43 @@ type IfaceType struct {
 	IfName string `json:"ifname"`
 }
 
+// simMode : Type indicating the desired simulation mode.
+type simMode string
+
+func (s *simMode) String() string {
+	return string(*s)
+}
+
+func (s *simMode) Set(value string) error {
+	switch value {
+	case "create":
+		fallthrough
+	case "create_continue":
+		fallthrough
+	case "delete":
+		*s = simMode(value)
+	default:
+		return fmt.Errorf("unknown sim mode %v", value)
+	}
+	return nil
+}
+
+func (s *simMode) create() bool {
+	return strings.Contains(string(*s), "create")
+}
+
+func (s *simMode) delete() bool {
+	return strings.Contains(string(*s), "delete")
+}
+
+func (s *simMode) keepGoing() bool {
+	return strings.Contains(string(*s), "continue")
+}
+
+func (s *simMode) enable() bool {
+	return string(*s) != ""
+}
+
```

Comment:
Move into grpcsim.go
Use enum type for simMode to do string compare only once during Set

Reply:
Makes sense, the string type was just for demonstration purposes.